pkg/pool-utils/contracts/lib/BasePoolMath.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+
+library BasePoolMath {
+    using FixedPoint for uint256;
+
+    function computeProportionalAmountsIn(
+        uint256[] memory balances,
+        uint256 bptTotalSupply,
+        uint256 bptAmountOut
+    ) internal pure returns (uint256[] memory amountsIn) {
+        /************************************************************************************
+        // computeProportionalAmountsIn                                                    //
+        // (per token)                                                                     //
+        // aI = amountIn                   /      bptOut      \                            //
+        // b = balance           aI = b * | ----------------- |                            //
+        // bptOut = bptAmountOut           \  bptTotalSupply  /                            //
+        // bpt = bptTotalSupply                                                            //
+        ************************************************************************************/
+
+        // Since we're computing amounts in, we round up overall. This means rounding up on both the multiplication and
+        //division.
```

Comment:
```suggestion
        // Since we're computing amounts in, we round up overall. This means rounding up on both the
        // multiplication and division.
```